Add comment about duplicate check in 'checkInvalidControlsAndCollectUnhandled' function within 'if' clause

Ahmad-S792 opened the pull request:
#### e7780d735de3975ec8bf854e4ed70d07c7648497
<pre>
Add comment about duplicate check in &apos;checkInvalidControlsAndCollectUnhandled&apos; function within &apos;if&apos; clause

<a href="https://bugs.webkit.org/show_bug.cgi?id=259010">https://bugs.webkit.org/show_bug.cgi?id=259010</a>

Reviewed by Tim Nguyen.

This patch just add explanatory comment of why there is &apos;duplicate&apos; check within &apos;if&apos; clause.

* Source/WebCore/html/HTMLFormElement.cpp:
(HTMLFormElement::checkInvalidControlsAndCollectUnhandled):

Canonical link: <a href="https://commits.webkit.org/265890@main">https://commits.webkit.org/265890@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1c2b88fd8771c77511399769b9106625254f38ae

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/12171 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/12516 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/12818 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/13916 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/11744 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/12201 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/14932 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/12531 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/14424 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/12335 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/13139 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/10306 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/14332 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/10424 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/11047 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/18160 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/11505 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/11207 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/14384 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/11699 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/9647 "1 flakes 1 failures") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/10916 "Built successfully") | | | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/2984 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/15242 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1360 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/11553 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (changes requested):
Ah the test failures reveal that this is actually not a duplicate check.

`!control->checkValidity(&unhandledInvalidControls)` can change the value of `control->form()` it seems.

Ahmad-S792 committed:
Add comment about duplicate check in 'checkInvalidControlsAndCollectUnhandled' function within 'if' clause

https://bugs.webkit.org/show_bug.cgi?id=259010

Reviewed by Tim Nguyen.

This patch just add explanatory comment of why there is 'duplicate' check within 'if' clause.

* Source/WebCore/html/HTMLFormElement.cpp:
(HTMLFormElement::checkInvalidControlsAndCollectUnhandled):

Canonical link: https://commits.webkit.org/265890@main
